pkgs/by-name/xg/xgen/package.nix

```
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+
+buildGoModule rec {
+  pname = "xgen";
+
+  # This is a pseudoversion including the fetch date and git hash.
+  # The upstream project does not provide official releases or version tags.
+  version = "v0-unstable-2023-07-02";
```

```suggestion
  version = "0-unstable-2023-07-02";
```

The `0-unstable-` versioning is well-established and does not require explaining. Additionally, according to the naming guidelines, the `version` MUST start with a number. You cannot prefix said number with a `v`.